mobile/lib/shared/ui/immich_image.dart

```diff
@@ -48,67 +57,22 @@ class ImmichImage extends StatelessWidget {
         ),
       );
     }
+
     final Asset asset = this.asset!;
-    if (useLocal(asset)) {
-      return Image(
-        image: localImageProvider(asset, size: preferredLocalAssetSize),
-        width: width,
-        height: height,
-        fit: fit,
-        frameBuilder: (context, child, frame, wasSynchronouslyLoaded) {
-          if (wasSynchronouslyLoaded || frame != null) {
-            return child;
-          }
 
-          // Show loading if desired
-          return Stack(
-            children: [
-              if (useGrayBoxPlaceholder)
-                const SizedBox.square(
-                  dimension: 250,
-                  child: DecoratedBox(
-                    decoration: BoxDecoration(color: Colors.grey),
-                  ),
-                ),
-              if (useProgressIndicator)
-                const Center(
-                  child: CircularProgressIndicator(),
-                ),
-            ],
-          );
-        },
-        errorBuilder: (context, error, stackTrace) {
-          if (error is PlatformException &&
-              error.code == "The asset not found!") {
-            debugPrint(
-              "Asset ${asset.localId} does not exist anymore on device!",
-            );
-          } else {
-            debugPrint(
-              "Error getting thumb for assetId=${asset.localId}: $error",
-            );
-          }
-          return Icon(
-            Icons.image_not_supported_outlined,
-            color: context.primaryColor,
-          );
-        },
-      );
-    }
-    final String? accessToken = Store.get(StoreKey.accessToken);
-    final String thumbnailRequestUrl = getThumbnailUrl(asset, type: type);
-    return CachedNetworkImage(
-      imageUrl: thumbnailRequestUrl,
-      httpHeaders: {"x-immich-user-token": accessToken ?? ""},
-      cacheKey: getThumbnailCacheKey(asset, type: type),
+    return Image(
+      image: ImmichImage.imageProvider(
+        asset: asset,
+        isThumbnail: isThumbnail,
+      ),
       width: width,
       height: height,
-      // keeping memCacheWidth, memCacheHeight, maxWidthDiskCache and
-      // maxHeightDiskCache = null allows to simply store the webp thumbnail
-      // from the server and use it for all rendered thumbnail sizes
       fit: fit,
-      fadeInDuration: const Duration(milliseconds: 250),
-      progressIndicatorBuilder: (context, url, downloadProgress) {
+      frameBuilder: (context, child, frame, wasSynchronouslyLoaded) {
```

Comment:
`CachedNetworkImage` handled fade in implicitly so when we scroll through the timeline, the assets used to fade in during loading. Since we switched to a custom handling, this is lost now. We have to handle this in `frameBuilder` or `loadingBuilder`